commit 106efddbe0e107b27efae789b6eb05dfb13eb94e
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-04-09T00:24:34+00:00 (6 months ago)
    
    TRITON-1373 cns-updater should gracefully handle nics without mac addresses

diff --git a/lib/napi-legacy-filter.js b/lib/napi-legacy-filter.js
index 6a8a981..5f0d004 100644
--- a/lib/napi-legacy-filter.js
+++ b/lib/napi-legacy-filter.js
@@ -57,11 +57,23 @@ NAPILegacyFilter.prototype._transform = function (vm, enc, cb) {
 	assert.object(vm, 'vm');
 	assert.arrayOfObject(vm.nics, 'vm.nics');
 
+	var self = this;
+
 	var nicsToFix = [];
 	vm.nics.forEach(function (nic) {
 		if (typeof (nic.network_uuid) !== 'string' ||
 		    nic.network_uuid === '' ||
 		    (!Array.isArray(nic.ips) && !nic.ip)) {
+
+			if (typeof (nic.mac) !== 'string') {
+				self.log.warn({
+				    vm: vm.uuid,
+				    nic: nic
+				}, 'found nic with non-string mac address; ' +
+				    'ignoring');
+				return;
+			}
+
 			nicsToFix.push(nic);
 		}
 	});
@@ -72,7 +84,6 @@ NAPILegacyFilter.prototype._transform = function (vm, enc, cb) {
 		return;
 	}
 
-	var self = this;
 	vasync.forEachParallel({
 		inputs: nicsToFix,
 		func: fixNic
